scripts/helpers.sh

```diff
@@ -62,3 +62,46 @@ command_exists() {
   local command="$1"
   command -v "$command" &> /dev/null
 }
+
+get_tmp_dir() {
+  local tmpdir="${TMPDIR:-${TMP:-${TEMP:-/tmp}}}"
+  [ -d "$tmpdir" ] || local tmpdir=~/tmp
+  echo "$tmpdir/tmux-$EUID-cpu"
+}
+
+get_time() {
+  date +%s.%N
+}
+
+get_cache_val(){
+  local key="$1"
+  # seconds after which cache is invalidated
+  local timeout="${2:-2}"
+  local cache="$(get_tmp_dir)/$key"
+  if [ -f "$cache" ]; then
+    awk -v cache="$(head -n1 "$cache")" -v timeout=$timeout -v now=$(get_time) \
+      'BEGIN {if (now - timeout < cache) exit 0; exit 1}' \
+      && tail -n+2 "$cache"
+  fi
+}
+
+put_cache_val(){
+  local key="$1"
+  local val="${@:2}"
+  local tmpdir="$(get_tmp_dir)"
+  [ ! -d "$tmpdir" ] && mkdir -p "$tmpdir" && chmod 0700 "$tmpdir"
```

Comment:
`chmod 0700` is not wanted on global system tmp directory

Reply:
it's never global. `get_tmp_dir` always returns a subdir.

Reply:
one of the main reasons I looked at `tmux-copycat`'s implementation for reference is because they would've fixed any problems with it by now if it was wrong.